src/main/features/response/views/timeline.njk

```diff
@@ -0,0 +1,93 @@
+  {% set error2 = form.errorFor('rows[' + index + '][description]').length > 0 %}
+  {% set displayTotalError = totalError and loop.index0 == 0 %}
+
+  <div class="timeline-row {% if error %} form-group-error {% endif %}">
+    {% if displayTotalError %}
+      <span class="error-message" id="rows[label]">{{ form.errorFor('rows') }}</span>
+    {% endif %}
+    {% if error1 %}
+      <span class="error-message" id="{{ 'rows[' + index + '][date][label]' }}">
+        {{ form.errorFor('rows[' + index + '][date]') }}
+      </span>
+    {% endif %}
+    {% if error2 %}
+      <span class="error-message" id="{{ 'rows[' + index + '][description][label]' }}">
+        {{ form.errorFor('rows[' + index + '][description]') }}
+      </span>
+    {% endif %}
+
+    <label class="form-label-bold mobile-show" for="timeline-date-{{ index }}">{{ index + 1 }}. Date</label>
+    <span class="form-hint mobile-show {% if error1 %}form-control-error{% endif %}">For example, 1 May 2017 </span>
+    <input class="form-control timeline-column-1" type="text" id="timeline-date-{{ index }}" name="timeline-date-{{ index }}" value="{{ form.valueFor(rows[' + index + '][date]) }}">
+
+    <label for="timeline-description-{{ index }}" class="form-label-bold mobile-show {% if error2 %}form-control-error{% endif %}">What happened</label>
+    <span class="form-hint mobile-show">For example, you might have signed a contract</span>
+    <textarea class="timeline-column-text" name="timeline-description-{{ index }}" id="timeline-description-{{ index }}" rows="2">{{ form.valueFor(rows[' + index + '][description]) }}</textarea>
+  </div>
+{% endmacro %}
+
+{% block content %}
+<div class="grid-row">
+  <div class="column-two-thirds">
+
+    <form method="post" novalidate>
+      {{ csrfProtection(csrf) }}
+      <h3 class="heading-small">{{ t('Add your timeline of events (optional)') }}</h3>
+      <p>{{ t('Include dates if you know them.') }}</p>
+      <div class="timeline">
+        <div class="timeline-headings">
+          <div id="timeline-heading-1" class="timeline-column-1">
+            <label class="form-label-bold" for="date-form-control">
+              {{ t('Date') }}
+            </label>
+            <span class="form-hint">{{ t('For example, 1 May 2017') }}</span>
+          </div>
+          <div id="timeline-heading-2" class="timeline-column-2">
+            <label class="form-label-bold" for="date-form-control">
+              {{ t('What happened') }}
+            </label>
+            <span class="form-hint">{{ t('For example, you might have signed a contract') }}</span>
+          </div>
+        </div>
+        <div class="timeline-rows">
+          {% set n = form.model.rows.length %}
+          {% if n %}
+            {% for row in form.model.rows %}
+              {{ timelineRow(loop.index0, form) }}
+            {% endfor %}
+          {% else %}
+            {{ timelineRow(0, form) }}
```

Comment:
this seems yuck, can't we just initialise it to 4?

Reply:
refactored